apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Footer/HostSidePanel/HostSidePanel.tsx

```diff
@@ -60,22 +89,39 @@ export function HostSidePanel(): ReactElement {
   }
 
   const handleSelectHost = async (hostId: string): Promise<void> => {
-    setSelectedHost(data.hosts.find((host) => host.id === hostId))
+    setSelectedHost(teamHosts.hosts.find((host) => host.id === hostId))
     setOpenSelect(false)
   }
 
   return (
     <>
       {/* DefaultHostPanel - no host */}
       <SidePanel title={t('Hosted By')} open edit>
-        {!openSelect && selectedHost == null && (
-          <SidePanelContainer border={false}>
-            <ContainedIconButton
-              label={t('Select a Host')}
-              thumbnailIcon={<UserProfiles2Icon />}
-              onClick={() => setOpenSelect(true)}
-            />
-          </SidePanelContainer>
+        {!openSelect && (selectedHost == null || !userInTeam) && (
+          <>
+            <SidePanelContainer border={!userInTeam}>
+              <ContainedIconButton
+                label={t('Select a Host')}
+                disabled={!userInTeam}
```

Comment:
This is always false right?
```suggestion
                disabled={false}
```

Reply:
We need to prevent the button from being clicked if the current user isn't in the same team as the journey. This is the whole point of the PR.

Reply:
Sorry this was meant to be true, but makes sense now